include/deal.II/distributed/tria.h

```diff
@@ -943,6 +940,172 @@ namespace parallel
+         * consists of the stem @p filename and an attached identifier
+         * <tt>-fixed.data</tt>.
+         *
+         * All processors write into that file simultaneously via MPIIO.
+         * Each processor's position to write to will be determined
+         * from the provided @p parallel_forest.
+         *
+         * Data has to be previously packed with pack_data().
+         */
+        void
+        save(const typename dealii::internal::p4est::types<dim>::forest
+               *         parallel_forest,
+             const char *filename) const;
+
+        /**
+         * Transfer data from file system.
+         *
+         * The data will be read from separate file, whose name
+         * consists of the stem @p filename and an attached identifier
+         * <tt>-fixed.data</tt>. The @p n_attached_deserialize parameter
+         * is required to gather the memory offsets for each callback.
+         *
+         * All processors read from that file simultaneously via MPIIO.
+         * Each processor's position to read from will be determined
+         * from the provided @p parallel_forest.
+         *
+         * After loading, unpack_data() needs to be called to finally
+         * distribute data across the associated triangulation.
+         */
+        void
+        load(const typename dealii::internal::p4est::types<dim>::forest
+               *                parallel_forest,
+             const char *       filename,
+             const unsigned int n_attached_deserialize);
+
+        /**
+         * Maximum number of callback functions that have been
+         * called for packing data over all cells.
+         *
+         * Returns the size of the class member cumulative_sizes_fix.
+         * Data has to be previously packed with pack_data() to get
+         * a non-zero result.
+         */
+        unsigned int
+        num_packed_datas();
+
+        /**
+         * Clears all containers and associated data, and resets member
+         * values to their default state.
+         *
+         * Frees memory completely.
+         */
+        void
+        clear();
+
+      private:
+        MPI_Comm mpi_communicator;
+
+        /**
+         * Cumulative size in bytes that those functions that have called
+         * register_data_attach() want to attach to each cell. This number
+         * only pertains to fixed-sized buffers where the data attached to
+         * each cell has exactly the same size.
+         *
+         * The last entry of this container corresponds to the data size
+         * packed per cell in the fixed size buffer (which can be accessed
+         * calling <tt>cumulative_sizes_fix.back()</tt>).
+         */
+        std::vector<unsigned int> cumulative_sizes_fix;
+
+        /**
+         * Consecutive buffers designed for the fixed size transfer
+         * functions of p4est.
+         */
+        std::vector<char> src_data_fix, dest_data_fix;
```

Comment:
same here.

also use one declaration per variable, rather than putting them into the same line